libretroshare/src/pqi/authssl.cc

```diff
@@ -989,411 +1002,353 @@ X509 *AuthSSLimpl::SignX509ReqWithGPG(X509_REQ *req, long /*days*/)
-		diagnostic = RS_SSL_HANDSHAKE_DIAGNOSTIC_CERTIFICATE_NOT_VALID ;
-		return false;
-	}
-
-	/* extract CN for peer Id */
-#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
-	RsPgpId issuer(std::string(getX509CNString(x509->cert_info->issuer)));
-#else
-	RsPgpId issuer(std::string(getX509CNString(X509_get_issuer_name(x509))));
-#endif
+	RsPgpId issuer = sslcert::getCertIssuer(*x509);
 	RsPeerDetails pd;
-#ifdef AUTHSSL_DEBUG
-	std::cerr << "Checking GPG issuer : " << issuer.toStdString() << std::endl ;
-#endif
-	if (!AuthGPG::getAuthGPG()->getGPGDetails(issuer, pd)) {
-		std::cerr << "AuthSSLimpl::AuthX509() X509 NOT authenticated : AuthGPG::getAuthGPG()->getGPGDetails() returned false." << std::endl;
-		diagnostic = RS_SSL_HANDSHAKE_DIAGNOSTIC_ISSUER_UNKNOWN ;
+	if (!AuthGPG::getAuthGPG()->getGPGDetails(issuer, pd))
+	{
+		RsErr() << __PRETTY_FUNCTION__ << " X509 NOT authenticated : "
+		        << "AuthGPG::getAuthGPG()->getGPGDetails(" << issuer
+		        << ",...) returned false." << std::endl;
+		diagnostic = RS_SSL_HANDSHAKE_DIAGNOSTIC_ISSUER_UNKNOWN;
 		return false;
 	}
+	else
+		Dbg3() << __PRETTY_FUNCTION__ << " issuer: " << issuer << " found"
+		       << std::endl;
 
 	/* verify GPG signature */
-
 	/*** NOW The Manual signing bit (HACKED FROM asn1/a_sign.c) ***/
 
 #if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 	int (*i2d)(X509_CINF*, unsigned char**) = i2d_X509_CINF;
-	ASN1_BIT_STRING *signature = x509->signature;
-	X509_CINF *data = x509->cert_info;
+	ASN1_BIT_STRING* signature = x509->signature;
+	X509_CINF* data = x509->cert_info;
 #else
-	const ASN1_BIT_STRING *signature = NULL ;
-    const X509_ALGOR *algor2=NULL;
-
+	const ASN1_BIT_STRING* signature = nullptr;
+	const X509_ALGOR* algor2 = nullptr;
 	X509_get0_signature(&signature,&algor2,x509);
 #endif
 
-	uint32_t certificate_version = getX509RetroshareCertificateVersion(x509) ;
+	uint32_t certificate_version = getX509RetroshareCertificateVersion(x509);
 
-	EVP_MD_CTX *ctx = EVP_MD_CTX_create();
-	int inl=0;
+	EVP_MD_CTX* ctx = EVP_MD_CTX_create();
+	int inl = 0;
 
-	const unsigned char *signed_data = NULL ;
-	uint32_t signed_data_length =0;
+	const unsigned char* signed_data = nullptr;
+	uint32_t signed_data_length = 0;
 
 	/* input buffer */
 #if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
-	inl=i2d(data,NULL);
-	unsigned char *buf_in=(unsigned char *)OPENSSL_malloc((unsigned int)inl);
-	unsigned char *p=NULL;
+	inl = i2d(data, nullptr);
+	unsigned char* buf_in = static_cast<unsigned char *>(
+	            OPENSSL_malloc(static_cast<unsigned int>(inl)) );
+	unsigned char* p = nullptr;
 #else
-	unsigned char *buf_in=NULL;
+	unsigned char* buf_in = nullptr;
 	inl=i2d_re_X509_tbs(x509,&buf_in) ;	// this does the i2d over x509->cert_info
 #endif
 
-	if(buf_in == NULL)
+	if(buf_in == nullptr)
```

Comment:
it's the same here: replaceing all NULL by nullptr is seriously not useful at all and wastes the time of the reviewer.

Reply:
It just avoid extra compiler warnings, so hopefully at some point we get to 0 or a number so small we easily notice when adding some code we introduce a new warning.